portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,70 +27,107 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+ 
 /** Servlet that returns a random quote. */
 @WebServlet("/data")
 public final class DataServlet extends HttpServlet {
+ 
+    /** Declared as class variable so that initialization happens once in init()
+        rather than everytime in doGET() */
+    List<String> funFacts = new ArrayList<>();
     
-    /** One time initialization may be used for future. */
-    // @Override
-    // public void init() {
-    // }
-
-
-    /** Merges dataFactsAndComments with pastComments to convert as one ArrayList into JSON to put on servlet. */
+    /** One time initialization. */
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        List<String> funFacts = new ArrayList<>();
+    public void init() {
         funFacts.add("I enjoy all music genres especially EDM!");
         funFacts.add("I studied abroad in Barcelona :)");
         funFacts.add("My favorite food is rotisserie chicken!");
         funFacts.add("I spend a lot of days on the lake!");
+    }
+ 
+    /** Merges dataFactsAndComments with pastComments to convert as one ArrayList into JSON to put on servlet. */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {        
+        // Add fun facts first to factsAndComments return variable.
+        List<String> factsAndComments = new ArrayList<>();
+        factsAndComments.addAll(funFacts);        
         
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         Query query = new Query("siteComments").addSort("timeStamp", SortDirection.DESCENDING);
         PreparedQuery results = datastore.prepare(query);
-
+ 
         List<String> pastComments = new ArrayList<>();
+        int validNumRequest = 0;
+        int numCommentsWanted = 0;
+        boolean numCommentsWantedTrigger = true;
+ 
         for (Entity entity : results.asIterable()) {
-            String comment = (String) entity.getProperty("newComment");
-            pastComments.add(comment);
+            validNumRequest = Integer.parseInt((String)(entity.getProperty("seeNumComments")));
+            if (validNumRequest >= 0){
+                String comment = (String) (entity.getProperty("newComment"));
+
+                //For testing queried result comments
+                // System.out.println("Query result comment: " + comment);
+
+                pastComments.add(comment);
+            } else {
+                System.out.println("Input of number comments wanted not greater than or equal to 0: " + validNumRequest);
```

Comment:
Is this supposed to be an error case?  If so, would it make sense to throw an exception in this case?  Obviously, that'd cause the page to fail to load, so perhaps that's the wrong choice - I don't have a good sense of what validNumRequest is meant to be.

Reply:
Yea from my knowledge I should do a print statement rather than error to keep the site going. I changed the validNumRequest variable name to validNumInput and added an explanation:
// validNumInput specifies if the form input for number of comments wanted is valid
                // (meaning >= 0). If not, then the comment entered in that entity won't be shown on the site.